src/optimizer/count_window_elimination.cpp

```diff
@@ -0,0 +1,271 @@
+				return op;
+			}
+
+			// --- Transformation ---
+
+			auto original_child = std::move(window.children[0]);
+			auto copy_child = original_child->Copy(optimizer.context);
+
+			// Rebind copy_child to avoid duplicate table indices
+			TableRebinder rebinder(optimizer);
+			rebinder.VisitOperator(*copy_child);
+
+			auto aggregate_index = optimizer.binder.GenerateTableIndex();
+			auto group_index = optimizer.binder.GenerateTableIndex();
+
+			vector<unique_ptr<Expression>> groups;
+			vector<unique_ptr<Expression>> aggregates;
+
+			// Create Aggregate Operator
+			for (auto &part : w_expr.partitions) {
+				auto part_copy = part->Copy();
+				rebinder.VisitExpression(&part_copy); // Update bindings
+				groups.push_back(std::move(part_copy));
+			}
+
+			auto count_func = *w_expr.aggregate;
+			unique_ptr<FunctionData> bind_info;
+			if (w_expr.bind_info) {
+				bind_info = w_expr.bind_info->Copy();
+			} else {
+				bind_info = nullptr;
+			}
+
+			vector<unique_ptr<Expression>> children;
+			for (auto &child : w_expr.children) {
+				auto child_copy = child->Copy();
+				rebinder.VisitExpression(&child_copy); // Update bindings
+				children.push_back(std::move(child_copy));
+			}
+
+			auto aggr_type = w_expr.distinct ? AggregateType::DISTINCT : AggregateType::NON_DISTINCT;
+
+			auto agg_expr = make_uniq<BoundAggregateExpression>(std::move(count_func), std::move(children), nullptr,
+			                                                    std::move(bind_info), aggr_type);
+
+			aggregates.push_back(std::move(agg_expr));
+
+			// args: group_index, aggregate_index, ...
+			auto agg_op = make_uniq<LogicalAggregate>(group_index, aggregate_index, std::move(aggregates));
+
+			agg_op->groups = std::move(groups);
+			agg_op->children.push_back(std::move(copy_child));
+			agg_op->ResolveOperatorTypes();
+
+			if (agg_op->types.size() <= agg_op->groups.size()) {
+				throw InternalException("LogicalAggregate types size mismatch");
+			}
+
+			// Filter on aggregate: count = 1
+			// Count is the first aggregate, so it's at agg_op->groups.size() in the types list
+			// Bindings: Aggregates are at aggregate_index
+			auto cnt_ref = make_uniq<BoundColumnRefExpression>(agg_op->types[agg_op->groups.size()],
+			                                                   ColumnBinding(aggregate_index, 0));
+
+			auto filter_expr =
+			    make_uniq<BoundComparisonExpression>(ExpressionType::COMPARE_EQUAL, std::move(cnt_ref),
+			                                         make_uniq<BoundConstantExpression>(Value::BIGINT(1)));
+
+			auto rhs_filter = make_uniq<LogicalFilter>();
+			rhs_filter->expressions.push_back(std::move(filter_expr));
+			rhs_filter->children.push_back(std::move(agg_op));
+			rhs_filter->ResolveOperatorTypes();
+
+			// Semi Join
+			auto join = make_uniq<LogicalComparisonJoin>(JoinType::SEMI);
+
+			for (size_t i = 0; i < w_expr.partitions.size(); ++i) {
+				JoinCondition cond;
+				cond.comparison = ExpressionType::COMPARE_EQUAL;
```

Comment:
I think you need `COMPARE_NOT_DISTINCT_FROM` here to handle the case of `NULL` keys?